File: java/src/main/java/com/ollang/sdk/resources/Revisions.java
package com.ollang.sdk.resources;

import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.ollang.sdk.OllangClient;

/** Request revisions on completed orders. */
public class Revisions {

  private final OllangClient client;

  public Revisions(OllangClient client) {
    this.client = client;
  }

  /**
   * Creates a revision request for an order.
   *
   * @param type revision type: one of {@code missingSubtitle}, {@code wrongSubtitle},
   *     {@code syncError}, {@code formatError} or {@code other}
   * @param time timestamp the revision refers to, e.g. {@code "00:01:23"}
   * @param description optional details for the reviewer; may be {@code null}
   */
  public JsonElement create(String orderId, String type, String time, String description) {
    JsonObject body = new JsonObject();
    body.addProperty("type", type);
    body.addProperty("time", time);
    if (description != null) {
      body.addProperty("description", description);
    }
    return client.post("/integration/revision/" + orderId, body);
  }

  public JsonElement list(String orderId) {
    return client.get("/integration/revision/" + orderId);
  }

  public JsonElement delete(String orderId, String revisionId) {
    return client.delete("/integration/revision/" + orderId + "/" + revisionId);
  }
}

File: python/src/ollang/resources/revisions.py
from typing import Any, Dict, List, Optional

from .._client import OllangClient


class Revisions:
    """Request revisions on completed orders."""

    def __init__(self, client: OllangClient):
        self._client = client

    def create(
        self,
        order_id: str,
        type: str,
        time: str,
        description: Optional[str] = None,
    ) -> Dict[str, Any]:
        """Create a revision request for an order.

        ``type`` is one of ``missingSubtitle``, ``wrongSubtitle``, ``syncError``,
        ``formatError`` or ``other``. ``time`` is the timestamp the revision
        refers to, e.g. ``"00:01:23"``.
        """
        body: Dict[str, Any] = {"type": type, "time": time}
        if description is not None:
            body["description"] = description
        return self._client.post(f"/integration/revision/{order_id}", json=body)

    def list(self, order_id: str) -> List[Dict[str, Any]]:
        return self._client.get(f"/integration/revision/{order_id}")

    def delete(self, order_id: str, revision_id: str) -> Any:
        return self._client.delete(f"/integration/revision/{order_id}/{revision_id}")
